common/src/main/java/org/apache/nemo/common/pass/Pass.java

```diff
@@ -27,7 +27,7 @@
  * Abstract class for optimization passes. All passes basically extends this class.
  */
 public abstract class Pass implements Serializable {
-  private Predicate<IRDAG> condition;
+  private Predicate<IRDAG> condition; //NOSONAR
```

Comment:
For this, let's leave this line as-is, and remove the Serializable tag on line 29. This makes more sense. Please remove the `implements Serializable` on line 29, and revert the change on line 30. This will solve the issue.